### PR TITLE
Use ThemeColor and add support for light themes

### PR DESCRIPTION
**Pull request by lnicola**
Part of #1294.

- switch to `ThemeColor`
- add light and high contrast theme definitions
- highlight control flow keywords and `unsafe`

### CODE REVIEW

File: editors/code/package.json [hunk 1]
```diff
@@ -268,6 +268,125 @@
...
+                "id": "ralsp.keyword",
+                "description": "Color for keywords",
+                "defaults": {
+                    "dark": "#F0DFAF",
+                    "light": "#0000FF",
+                    "highContrast": "#569CD6"
+                }
+            },
+            {
+                "id": "ralsp.control",
```
Comment (matklad) [inline editors/code/package.json]: Is there an easy way to allow both

* highligting all keywords the same (this is what I personally prefer)
* highlight various different classes of keywords differently

Perhaps we should just name classes `ralsp.keyword`, `ralsp.keyword.controlflow`, `ralsp.keyword.unsafe`?

File: editors/code/package.json [hunk 2]
```diff
@@ -268,6 +268,125 @@
...
+                    "dark": "#FF3030",
+                    "light": "#FF1010",
+                    "highContrast": "#FF1010"
+                }
+            },
+            {
+                "id": "ralsp.keyword",
+                "description": "Color for keywords",
+                "defaults": {
+                    "dark": "#F0DFAF",
```
Comment (matklad) [inline editors/code/package.json]: FWIW, I'd be perfectly fine with picking a color from default dark theme. This is now configurable, so I can just specify zenburn colors in my config

File: crates/ra_ide_api/src/syntax_highlighting.rs [hunk 3]
```diff
@@ -11,6 +11,14 @@ pub struct HighlightedRange {
     pub tag: &'static str,
 }
 
+fn is_control_keyword(kind: SyntaxKind) -> bool {
+    match kind {
+        FOR_KW | LOOP_KW | WHILE_KW | CONTINUE_KW | BREAK_KW | IF_KW | ELSE_KW | MATCH_KW
```
Comment (matklad) [inline crates/ra_ide_api/src/syntax_highlighting.rs]: The new syntax for this is

```
match kind {
    T![for], T![loop], ...
}
```

File: editors/code/package.json (hunk 1 above)
Comment (lnicola) [reply editors/code/package.json]: Updated. I used `ralsp.keyword.control` to match the default theme (`keyword.control`).

File: editors/code/package.json (hunk 2 above)
Comment (lnicola) [reply editors/code/package.json]: I switched to the default colors.

File: crates/ra_ide_api/src/syntax_highlighting.rs (hunk 3 above)
Comment (lnicola) [reply crates/ra_ide_api/src/syntax_highlighting.rs]: Is that better? :sweat_smile:

Updated.

Comment (matklad) [reply crates/ra_ide_api/src/syntax_highlighting.rs]: I think it is, but the beauty is in the eye of the beholder :)

**Comment by matklad** [comment]
Thanks!

bors r+

File: editors/code/package.json [hunk 4]
```diff
@@ -268,6 +268,116 @@
...
+                "description": "Color for literals",
+                "defaults": {
+                    "dark": "#BECEA8",
+                    "light": "#09885A",
+                    "highContrast": "#B5CEA8"
+                }
+            },
+            {
+                "id": "ralsp.macro",
+                "description": "Color for DFAF8F",
```
Comment (mominul) [inline editors/code/package.json]: Wouldn't be it `Color for macros`?

Comment (lnicola) [reply editors/code/package.json]: Thank you, fixed in https://github.com/rust-analyzer/rust-analyzer/pull/1305.

**Pull request merged**